src/rtpghi~.c

```diff
@@ -120,39 +102,58 @@ void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
     else {
         post("failed to init, status %d", init_s);
     }
+   
+   x->c = getbytes(M * sizeof *(x->c));
+   post("length of c[]: %d", (M/2+1) * sizeof *(x->c));
+   post("s_n: %d", sp[0]->s_n);
+   
+   dsp_add(rtpghi_tilde_perform, 3,x, sp[0]->s_vec, sp[1]->s_vec, sp[2]->s_vec, sp[0]->s_n);
+}
+
+void *rtpghi_tilde_new(t_symbol *s, int argc, t_atom *argv)
+{
+
+  t_rtpghi_tilde *x = (t_rtpghi_tilde *)pd_new(rtpghi_tilde_class);
+
+    x->ol_pd=atom_getfloat(argv);
+    x->M_pd=atom_getfloat(argv+1);
```

Comment:
you must check the number of arguments before accessing them.

something like

```C
switch (argc) {
  case 5: default:
     x->window_type=atom_getsymbol(argv+4);
  /* fallthrough */
  case 4:
    x->do_causal_pd=atom_getfloat(argv+3);
  /* fallthrough */
  case 3:
  // ...
}
```

alternatively you could use `atom_getfloatarg()`, but then you cannot assign default values != 0.+